### DISCUSSION TITLE
An option specifically for denying warnings

### DISCUSSION THREAD

**Discussion by mightyiam** [90-008/nix-cargo-integration#174] [Ideas]
Following up on https://github.com/yusdacra/nix-cargo-integration/discussions/157#discussion-6925315. Perhaps it would fit within the abstraction level of this library to provide an option for denying warnings? When enabled, it would append `--deny warnings` to both `RUSTFLAGS` and `RUSTDOCFLAGS`.

Possibly `perSystem.nci.projects.<name>.denyWarnings` and `perSystem.nci.crates.<name>.denyWarnings`?

**Comment by 90-008** [comment]
i don't think this is necessary, i'd probably add a `rustFlags` option instead to bring `drvConfig.env.RUSTFLAGS` up

**Reply by mightyiam** [reply]
It seems such a common preference. During development warnings are allowed. But building with Nix... That's probably where we wanna deny warnings. Usually. Like; who's going to complain about it even being default instead of fixing their warnings? This is Rust, after all.

**Reply by 90-008** [reply]
it's a very small addition but these do add up, i don't want to have a million small options. it is already well known that you can use RUSTFLAGS etc. to configure these so i'd rather add an option that works in tandem with what's already known, instead of adding very specific options each time someone asks for one (unless there is a good reason for it, i don't think this is one of those cases though).
